diff -rupN a/configure.ac b/configure.ac
--- a/configure.ac	2015-07-21 19:50:11.000000000 +0000
+++ b/configure.ac	2015-10-10 07:07:47.667072355 +0000
@@ -973,16 +973,16 @@ AC_CHECK_FUNCS([eventfd])
 
 dnl ** Check for __thread support in the compiler
 AC_MSG_CHECKING(for __thread support)
-AC_COMPILE_IFELSE(
-  [ AC_LANG_SOURCE([[__thread int tester = 0;]]) ],
-  [
-   AC_MSG_RESULT(yes)
-   AC_DEFINE([CC_SUPPORTS_TLS],[1],[Define to 1 if __thread is supported])
-  ],
-  [
+#AC_COMPILE_IFELSE(
+#  [ AC_LANG_SOURCE([[__thread int tester = 0;]]) ],
+#  [
+#   AC_MSG_RESULT(yes)
+#   AC_DEFINE([CC_SUPPORTS_TLS],[1],[Define to 1 if __thread is supported])
+#  ],
+#  [
    AC_MSG_RESULT(no)
    AC_DEFINE([CC_SUPPORTS_TLS],[0],[Define to 1 if __thread is supported])
-  ])
+#  ])
 
 
 dnl ** checking for PAPI
