diff -rupN a/compiler/main/DriverPipeline.hs b/compiler/main/DriverPipeline.hs
--- a/compiler/main/DriverPipeline.hs	2015-07-21 13:52:50.000000000 +0000
+++ b/compiler/main/DriverPipeline.hs	2015-10-10 22:26:05.106785648 +0000
@@ -1432,7 +1432,7 @@ runPhase (RealPhase LlvmLlc) input_fn df
         rmodel | platformOS (targetPlatform dflags) == OSiOS = "dynamic-no-pic"
                | gopt Opt_PIC dflags                         = "pic"
                | not (gopt Opt_Static dflags)                = "dynamic-no-pic"
-               | otherwise                                   = "static"
+               | otherwise                                   = "pic" -- "static"
         tbaa | ver < 29                 = "" -- no tbaa in 2.8 and earlier
              | gopt Opt_LlvmTBAA dflags = "--enable-tbaa=true"
              | otherwise                = "--enable-tbaa=false"
