diff -rupN a/rts/RtsMessages.c b/rts/RtsMessages.c
--- a/rts/RtsMessages.c	2014-12-23 02:31:10.000000000 +0000
+++ b/rts/RtsMessages.c	2015-10-11 22:25:49.802309919 +0000
@@ -19,6 +19,11 @@
 #include <windows.h>
 #endif
 
+#ifdef __ANDROID__
+#include <dlfcn.h>
+#include <android/log.h>
+#endif
+
 /* -----------------------------------------------------------------------------
    General message generation functions
 
@@ -163,6 +168,26 @@ rtsFatalInternalErrorFn(const char *s, v
      fflush(stderr);
   }
 
+#ifdef __ANDROID__
+  {
+      /* don't fflush(stdout); WORKAROUND bug in Linux glibc */
+      void* handle = dlopen("liblog.so", RTLD_GLOBAL | RTLD_NOW);
+      int (*__android_log_write)(int, const char*, const char*) = dlsym(handle, "__android_log_write");
+  //    if (prog_argv != NULL && prog_name != NULL) {
+  //        char* s1;
+  //        sprintf(s1, "%s: internal error: ", prog_name);
+  //        __android_log_write( ANDROID_LOG_ERROR, "ghc-rts", s1); 
+  //    } else {
+  //        char* s1;
+  //        sprintf(s1, "internal error: ");
+  //        __android_log_write( ANDROID_LOG_ERROR, "ghc-rts", s1);
+  //    }
+      char* s2;
+      vsprintf(s2, s, ap);
+      __android_log_write( ANDROID_LOG_ERROR, "ghc-rts", s2);
+  }
+#endif // __ANDROID__
+   
 #ifdef TRACING
   if (RtsFlags.TraceFlags.tracing == TRACE_EVENTLOG) endEventLogging();
 #endif
